### ISSUE TITLE
[expo-task-manager] WARN  `new NativeEventEmitter()` was called with a non-null argument without the required `removeListeners` method.

### ISSUE THREAD

**Issue by mgscreativa** [expo/expo#26600]
### Minimal reproducible example

none

### Summary

Upgraded to SDK 50 then (I guess) expo-task-manager started to get this warnings

expo-task-manager: ~11.7.0

```
 WARN  `new NativeEventEmitter()` was called with a non-null argument without the required `removeListeners` method.
 WARN  `new NativeEventEmitter()` was called with a non-null argument without the required `addListener` method.
 WARN  `new NativeEventEmitter()` was called with a non-null argument without the required `removeListeners` method.
 WARN  `new NativeEventEmitter()` was called with a non-null argument without the required `addListener` method.
 WARN  `new NativeEventEmitter()` was called with a non-null argument without the required `removeListeners` method.
 WARN  `new NativeEventEmitter()` was called with a non-null argument without the required `addListener` method.
 WARN  `new NativeEventEmitter()` was called with a non-null argument without the required `removeListeners` method.
```

### See screenshot
![Screenshot_1705941770](https://github.com/expo/expo/assets/1286632/3615897e-5896-464c-9179-2ad8a5ca2eef)

### setupNotificationsBackgroundTask.js code:
```
import { useEffect } from 'react';
import * as TaskManager from 'expo-task-manager';
import * as Notifications from 'expo-notifications';
import { Platform } from 'react-native';

import config from '../config';
import { checkExperienceId, sanitizeNotificationUrl } from '../helpers';
import { notificationsPersistState } from '../stores';

let addNotificationFn = () => {
  console.log('[Notifications Back] addNotification not yet initialized');
};

let checkExperienceIdFn = () => {
  console.log('[Notifications Back] checkExperienceId not yet initialized');
};

let sanitizeNotificationUrlFn = () => {
  console.log('[Notifications Back] checkExperienceId not yet initialized');
};

let getUnreadNotificationsFn = () => {
  console.log('[Notifications Back] getUnreadNotifications not yet initialized');
};

const BACKGROUND_NOTIFICATION_TASK = 'BACKGROUND-NOTIFICATION-TASK';
TaskManager.defineTask(BACKGROUND_NOTIFICATION_TASK, ({ data, error, executionInfo }) => {
  console.log(`[Notifications Back] Background notification received! ${JSON.stringify(data)}`);

  let notification = {};
  if (Platform.OS === 'android') {
    notification = {
      request: {
        trigger: {
          remoteMessage: {
            data: {
              experienceId: data.notification.data.experienceId,
            },
          },
        },
      },
    };
  } else if (Platform.OS === 'ios') {
    notification = {
      request: {
        trigger: {
          payload: {
            experienceId: data.experienceId,
          },
        },
      },
    };
  }

  console.log(`[Notifications Back] checkExperienceIdFn notification: ${JSON.stringify(notification)}`);

  let message = {};
  if (checkExperienceIdFn(notification)) {
    if (Platform.OS === 'android') {
      data.notification.data.body = JSON.parse(data.notification.data.body);

      message = {
        msgId: data.notification.data.body.msgId.trim(),
        title: data.notification.data.title,
        message: data.notification.data.message,
        link:
          data.notification.data.body.link !== null
            ? sanitizeNotificationUrlFn(data.notification.data.body.link.trim())
            : '',
        read: false,
      };
    } else if (Platform.OS === 'ios') {
      message = {
        msgId: data.body.msgId.trim(),
        title: data.aps.alert.title,
        message: data.aps.alert.body,
        link: data.body.link !== null ? sanitizeNotificationUrlFn(data.body.link.trim()) : '',
        read: false,
      };
    }

    console.log(`[Notifications Back] Notification message compiled: ${JSON.stringify(message)}`);
    console.log(`[Notifications Back] Trying to add msgId: ${message.msgId}`);

    addNotificationFn(message);

    if (config.setAndroidIconBadge || Platform.OS === 'ios') {
      const unreadNotifications = getUnreadNotificationsFn();
      console.log(`[Notifications Back] Setting badge count to ${unreadNotifications}`);

      Notifications.setBadgeCountAsync(unreadNotifications)
        .then(() => null)
        .catch(() => null);
    }
  }
});

export default function setupNotificationsBackgroundTask() {
  const { addNotification, getUnreadNotifications } = notificationsPersistState(
    ({ addNotification, getUnreadNotifications }) => ({
      addNotification,
      getUnreadNotifications,
    }),
  );

  addNotificationFn = addNotification;
  checkExperienceIdFn = checkExperienceId;
  getUnreadNotificationsFn = getUnreadNotifications;
  sanitizeNotificationUrlFn = sanitizeNotificationUrl;

  useEffect(() => {
    registerBackgroundTask()
      .then(() => null)
      .catch(() => null);
  }, []);
}

async function registerBackgroundTask() {
  try {
    await Notifications.registerTaskAsync(BACKGROUND_NOTIFICATION_TASK);
  } catch (e) {
    console.log(e);
  }
}

```

### Environment

  expo-env-info 1.2.0 environment info:
    System:
      OS: Linux 5.15 Linux Mint 21.3 (Virginia)
      Shell: 5.1.16 - /bin/bash
    Binaries:
      Node: 18.19.0 - ~/.nvm/versions/node/v18.19.0/bin/node
      Yarn: 1.22.21 - ~/.nvm/versions/node/v18.19.0/bin/yarn
      npm: 10.2.3 - ~/.nvm/versions/node/v18.19.0/bin/npm
      Watchman: 20230430.125247.0 - /usr/local/bin/watchman
    npmPackages:
      expo: ~50.0.2 => 50.0.2 
      react: 18.2.0 => 18.2.0 
      react-dom: 18.2.0 => 18.2.0 
      react-native: 0.73.2 => 0.73.2 
      react-native-web: ~0.19.6 => 0.19.10 
    npmGlobalPackages:
      eas-cli: 7.0.0
    Expo Workflow: managed

**Comment by jerielng** [comment]
I believe this is being caused by `expo-notifications`. I'm seeing the same warnings when calling `Notifications.addNotificationReceivedListener` and `Notifications.addNotificationResponseReceivedListener`.

**Comment by mgscreativa** [comment]
Hi @alanjhughes do you have any ETA to get this fix published as package update?

**Comment by alanjhughes** [comment]
Hi @mgscreativa - fix is published in `11.7.2`
